llvm/lib/Target/EraVM/EraVMInstrInfo.td

```diff
@@ -1248,3 +1259,95 @@ def : Pat<(EraVMTrap), (PANICr R0)>;
 
 def : Pat<(EraVMselectcc GRPTR:$rs0, GRPTR:$rs1, imm:$cc),
           (FATPTR_SELrrr GRPTR:$rs0, GRPTR:$rs1, imm:$cc)>;
+
+// ============================ Overflow ADDs =====================================
+// to register
+def : Pat<(EraVMAdd_v GR256:$rs0, GR256:$rs1),
+          (ADDrrr_v GR256:$rs0, GR256:$rs1, 0)>;
+def : Pat<(EraVMAdd_v GR256:$rs0, imm16:$rs1),
+          (ADDirr_v imm16:$rs1, GR256:$rs0, 0)>;
+def : Pat<(EraVMAdd_v GR256:$rs0, (load_code memaddr:$rs1)),
+          (ADDcrr_v memop:$rs1, GR256:$rs0, 0)>;
+def : Pat<(EraVMAdd_v (load_code memaddr:$rs0), GR256:$rs1),
+          (ADDcrr_v memop:$rs0, GR256:$rs1, 0)>;
+def : Pat<(EraVMAdd_v GR256:$rs0, (load_stack stackaddr:$rs1)),
+          (ADDsrr_v stackin:$rs1, GR256:$rs0, 0)>;
+def : Pat<(EraVMAdd_v (load_stack stackaddr:$rs0), GR256:$rs1),
+          (ADDsrr_v stackin:$rs0, GR256:$rs1, 0)>;
+
+// to stack
+def : Pat<(store_stack (EraVMAdd_v GR256:$rs0, GR256:$rs1), stackaddr:$stack),
+          (ADDrrs_v GR256:$rs0, GR256:$rs1, stackaddr:$stack, 0)>;
+def : Pat<(store_stack (EraVMAdd_v GR256:$rs0, imm16:$rs1), stackaddr:$stack),
+          (ADDirs_v imm16:$rs1, GR256:$rs0, stackin:$stack, 0)>;
+def : Pat<(store_stack (EraVMAdd_v GR256:$rs0, (load_code memaddr:$rs1)), stackaddr:$stack),
+          (ADDcrs_v memop:$rs1, GR256:$rs0, stackin:$stack, 0)>;
+def : Pat<(store_stack (EraVMAdd_v (load_code memaddr:$rs0), GR256:$rs1), stackaddr:$stack),
+          (ADDcrs_v memop:$rs0, GR256:$rs1, stackin:$stack, 0)>;
+def : Pat<(store_stack (EraVMAdd_v (load_stack stackaddr:$rs0), GR256:$rs1), stackaddr:$stack),
+          (ADDsrs_v stackin:$rs0, GR256:$rs1, stackin:$stack, 0)>;
+def : Pat<(store_stack (EraVMAdd_v GR256:$rs0, (load_stack stackaddr:$rs1)), stackaddr:$stack),
+          (ADDsrs_v stackin:$rs1, GR256:$rs0, stackin:$stack, 0)>;
+
+// ============================ Overflow SUBs =====================================
```

Comment:
Patterns for large_imm are missing.

Reply:
`large_imm` is added now just like `imm16` .